packages/connection-storage/src/compass-renderer-connection-storage.ts

```diff
@@ -97,12 +106,17 @@ class CompassRendererConnectionStorage
   async getAutoConnectInfo(
     autoConnectPreferences?: AutoConnectPreferences
   ): Promise<ConnectionInfo | undefined> {
-    return await this.ipc.getAutoConnectInfo(
+    if (this.hasAlreadyRequestedAutoConnectInfo) {
+      return;
+    }
+    const autoConnectInfo = await this.ipc.getAutoConnectInfo(
       autoConnectPreferences ??
         (await this.getInitialAutoConnectPreferences?.()) ?? {
           shouldAutoConnect: false,
         }
     );
+    this.hasAlreadyRequestedAutoConnectInfo = true;
```

Comment:
This should probably be set syncronously before `ipc.getAutoConnectInfo` is called to make sure that async io doesn't introduce some race condition issue here

Reply:
a sad miss, my bad. Fixed it just now